lib/segment/src/index/struct_payload_index.rs

```diff
@@ -38,16 +37,20 @@ use crate::telemetry::PayloadIndexTelemetry;
 use crate::types::{
     infer_collection_value_type, infer_value_type, Condition, FieldCondition, Filter,
     IsEmptyCondition, IsNullCondition, Payload, PayloadContainer, PayloadField, PayloadFieldSchema,
-    PayloadKeyType, PayloadKeyTypeRef, PayloadSchemaType,
+    PayloadKeyType, PayloadKeyTypeRef, PayloadSchemaType, VectorName,
 };
+use crate::vector_storage::{VectorStorage, VectorStorageEnum};
 
 /// `PayloadIndex` implementation, which actually uses index structures for providing faster search
 #[derive(Debug)]
 pub struct StructPayloadIndex {
     /// Payload storage
     payload: Arc<AtomicRefCell<PayloadStorageEnum>>,
     /// Used for `has_id` condition and estimating cardinality
-    id_tracker: Arc<AtomicRefCell<IdTrackerSS>>,
+    pub(super) id_tracker: Arc<AtomicRefCell<IdTrackerSS>>,
+    /// Vector storages for each field, used for `has_vector` condition
+    #[allow(dead_code)]
```

Comment:
I guess the `#[allow(dead_code)]` can go now?

Reply:
:heavy_check_mark: